cmd/go-sbot/config_test.go

```diff
@@ -0,0 +1,166 @@
+	err := os.WriteFile(configPath, []byte(configContents), 0700)
+	r.NoError(err, "write config file")
+	configFromDisk := readConfig(configPath)
+	// config values should be read correctly
+	r.EqualValues(expectedConfig.Hops, configFromDisk.Hops)
+	r.EqualValues(expectedConfig.MuxRPCAddress, configFromDisk.MuxRPCAddress)
+	r.EqualValues(expectedConfig.WebsocketAddress, configFromDisk.WebsocketAddress)
+	r.EqualValues(expectedConfig.EnableAdvertiseUDP, configFromDisk.EnableAdvertiseUDP)
+	r.EqualValues(expectedConfig.EnableDiscoveryUDP, configFromDisk.EnableDiscoveryUDP)
+	r.EqualValues(expectedConfig.EnableEBT, configFromDisk.EnableEBT)
+	r.EqualValues(expectedConfig.EnableFirewall, configFromDisk.EnableFirewall)
+	r.EqualValues(expectedConfig.NoUnixSocket, configFromDisk.NoUnixSocket)
+}
+
+func TestUnmarshalConfig(t *testing.T) {
+	r := require.New(t)
+	config := SbotConfig{
+		NoUnixSocket: true,
+		EnableAdvertiseUDP: true,
+		EnableDiscoveryUDP: true,
+		EnableEBT: true,
+		EnableFirewall: true,
+		RepairFSBeforeStart: true,
+	}
+	b, err := json.MarshalIndent(config, "", "  ")
+	r.NoError(err)
+	configStr := string(b)
+	expectedValues := strings.Split(strings.TrimSpace(`"nounixsock": true,
+  "localadv": true,
+  "localdiscov": true,
+  "enable-ebt": true,
+  "promisc": true,
+  "repair": true
+	`), "\n")
+	for _, expected := range expectedValues {
+		r.True(strings.Contains(configStr, expected), expected)
+	}
+}
+
+func TestConfiguredSbot (t *testing.T) {
+	r := require.New(t)
+	configContents := `# Supply various flags to control go-sbot options.
+hops = 2 
+
+shscap = "0KHLiKZvAvjbY1ziZEHMXawbCEIM6qwjCDm3VYRan/s=" 
+# Address to listen on
+lis = ":8008" 
+# Address to listen on for ssb websocket connections
+wslis = ":8989" 
+
+# Enable sending local UDP broadcasts
+localadv = "on"
+# Enable connecting to incoming UDP broadcasts
+localdiscov = true
+# Enable syncing by using epidemic-broadcast-trees (EBT)
+enable-ebt = false 
+# Bypass graph auth and fetch remote's feed, useful for pubs that are restoring their data from peers. Caveats abound, however.
+promisc = false 
+# Disable the UNIX socket RPC interface
+nounixsock = false 
+`
+	expectedConfig := SbotConfig{
+		ShsCap: "0KHLiKZvAvjbY1ziZEHMXawbCEIM6qwjCDm3VYRan/s=",
+		Hops: 2,
+		MuxRPCAddress: ":8008",
+		WebsocketAddress: ":8989",
+		EnableAdvertiseUDP: true,
+		EnableDiscoveryUDP: true,
+		EnableEBT: false,
+		EnableFirewall: false,
+		NoUnixSocket: false,
+	}
+	testPath := filepath.Join(".", "testrun", t.Name())
+	r.NoError(os.RemoveAll(testPath), "remove testrun folder")
+	r.NoError(os.MkdirAll(testPath, 0700), "make new testrun folder")
+	configPath := filepath.Join(testPath, "config.toml")
+	err := os.WriteFile(configPath, []byte(configContents), 0700)
+	r.NoError(err, "write config file")
+
+	// now start an sbot with the config
+	binName := "go-sbot-testing"
+	binPath := filepath.Join(testPath, binName)
+
+	goBuild := exec.Command("go", "build", "-o", binPath)
```

Comment:
```suggestion
	goBuild := exec.Command("go", "build", "-o", binPath)
	goBuild.Stderr = os.Stderr
	goBuild.Stdout = os.Stderr
```

just so that we are not treated with an empty error if it failes

Reply:
:+1: